File: src/Services/Project/TalentMesh.ProjectService/Models/Project.cs
namespace TalentMesh.ProjectService.Models
{
    public class Project
    {
        public Guid Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public string ClientName { get; set; } = string.Empty;

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public string Status { get; set; } = "Planning";

        public Guid CreatedBy { get; set; }

        public ICollection<ProjectSkillRequirement> SkillRequirements { get; set; }
            = new List<ProjectSkillRequirement>();
    }
}

File: src/Frontend/TalentMesh-UI/src/app/models/project.ts

export interface ProjectskillRequirements {
    id : string;
    skillName : string;
    requiredLevel : string;
    requiredCount : number;
}

export interface Project {
    id : string;
    name : string;
    description : string;
    startDate : string;
    endDate:  string;
    clientName : string;
    status : string;
    skillRequirements : ProjectskillRequirements[];
}